### PR TITLE
[pattgen, dv] Add initial pattgen_testplan.md for pattgen ip

### PR DESCRIPTION
**Pull request by tunghoang290780**
This PR adds an initial pattgen_testplan.md file for pattgen IP which is built upon on the latest specification PR #3235 

Signed-off-by: Tung Hoang <hoang.tung@wdc.com>

### CODE REVIEW

**Comment by tunghoang290780** [comment]
The latest specification document has been submitted with PR #3235 (under review)

File: hw/ip/pattgen/data/pattgen_testplan.hjson [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+            Stimulus:
+              - Randomly reset DUT
+              - Restart pattern generation
+
+            Checking:
+              - Ensure registers are properly reset
+            ''' 
+      milestone: V2
+      tests: ["pattgen_stress_all_with_rand_reset"]
```
Comment (cindychip) [inline hw/ip/pattgen/data/pattgen_testplan.hjson:96]: Do you plan to use standard `stress_all_with_rand_reset` , if so, maybe you can directly import from here: https://github.com/lowRISC/opentitan/blob/master/hw/dv/tools/testplans/stress_all_with_reset_testplan.hjson.

Comment (tunghoang290780) [reply hw/ip/pattgen/data/pattgen_testplan.hjson:96]: It should be standard `stress_all_with_rand_reset` but I think we still need to define that in the testplan even it requires zero-cost of development at IP level

Comment (cindychip) [reply hw/ip/pattgen/data/pattgen_testplan.hjson:96]: Thanks for confirming, if this is the standard stress_all_with_rand_reset, you can just import the testplan like the CSR_test_suites in line 6-8 :)

File: hw/ip/pattgen/data/pattgen_testplan.hjson [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+              - Check completion interrupts are asserted once a pattern
+                is completely generated on the active channels
+            '''
+      milestone: V1
+      tests: ["pattgen_sanity"]
+    }
+    {
+      name: intr_perf
+      desc: '''
+            Checking ip operation at min/max bandwith
```
Comment (weicaiyang) [inline hw/ip/pattgen/data/pattgen_testplan.hjson:36]: bandwith -> bandwidth

File: hw/ip/pattgen/data/pattgen_testplan.hjson [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    {
+      name: stress_all
+      desc: '''
+            Combine above sequences in one test to run sequentially,
+
+            Stimulus:
+              - Start sequences and randomly add reset between each sequence
+
+            Checking:
+              - All sequences should finished and checked by the scoreboard
```
Comment (weicaiyang) [inline hw/ip/pattgen/data/pattgen_testplan.hjson:77]: nit: 
```suggestion
              - All sequences should be finished and checked by the scoreboard
```

File: hw/ip/pattgen/data/pattgen_testplan.hjson (hunk 2 above)
Comment (tunghoang290780) [reply hw/ip/pattgen/data/pattgen_testplan.hjson:36]: Fixed

File: hw/ip/pattgen/data/pattgen_testplan.hjson (hunk 3 above)
Comment (tunghoang290780) [reply hw/ip/pattgen/data/pattgen_testplan.hjson:77]: fixed

**Comment by tunghoang290780** [comment]
@weicaiyang @cindychip just forced-push to address your comments

File: hw/ip/pattgen/data/pattgen_testplan.hjson (hunk 1 above)
Comment (tunghoang290780) [reply hw/ip/pattgen/data/pattgen_testplan.hjson]: Got your point, remove  the description for `stress_all_with_rand_reset` by importing `hw/dv/tools/testplans/stress_all_with_reset_testplan.hjson`

**Review by weicaiyang — APPROVED** [review]
LGTM

**Review by rasmus-madsen — APPROVED** [review]

**Comment by rasmus-madsen** [comment]
LGTM I will merge this

**Comment by rasmus-madsen** [comment]
Sorry I think I merged this prematurely.

**Comment by rasmus-madsen** [comment]
@tunghoang290780 
I am sorry but I merged this prematurely.
@cindychip should accept your changes first and we probably also need approval from the RTL designer on this.
could you please re push your last commit

**Pull request merged**